src/renderer/components/shared/CanvasDataBuilder.js

```diff
@@ -512,52 +512,50 @@ const buildNeighbours = async (targetConcept, answers) => {
  * @param {*} graknTx
  */
 const buildRPInstances = async (answers, currentData, shouldLimit, graknTx) => {
-  let edges = [];
-  const nodes = [];
-
-  for (let i = 0; i < answers.length; i += 1) {
-    const answer = answers[i];
-    const answersGroup = Array.from(answer.map().entries());
+  const getRolePlayersData = () => {
+    const promises = [];
+    const edges = [];
+    const nodes = [];
+
+    answers.forEach((answer) => {
```

Comment:
for a moment i thought this was too hard to read, but then i saw the old code hehehe

Reply:
yeah, I'm also finding it challenging to write this in a way that's a 100% non-blocking and at the same time keep it highly readable :/